sap-vcap-services/src/main/java/com/sap/cloud/environment/servicebinding/SapVcapServicesServiceBindingAccessor.java

```diff
@@ -68,11 +74,31 @@ public SapVcapServicesServiceBindingAccessor( @Nonnull final Function<String, St
     public List<ServiceBinding> getServiceBindings()
         throws ServiceBindingAccessException
     {
-        logger.debug("Trying to determine service bindings using the '{}' environment variable.", VCAP_SERVICES);
-        final String vcapServices = environmentVariableReader.apply(VCAP_SERVICES);
+        logger
+            .debug(
+                "Trying to determine service bindings using the '{}' and '{}' environment variables.",
+                VCAP_SERVICES,
+                VCAP_SERVICES_FILE_PATH);
+        String vcapServices = environmentVariableReader.apply(VCAP_SERVICES);
+        final String vcapServicesFilePath = environmentVariableReader.apply(VCAP_SERVICES_FILE_PATH);
+
+        if( vcapServices == null && vcapServicesFilePath != null ) {
+            logger
+                .debug(
+                    "Environment variable '{}' is not defined. Falling back to environment variable '{}'",
+                    VCAP_SERVICES,
+                    VCAP_SERVICES_FILE_PATH);
+            try {
+                vcapServices = Files.readString(Path.of(vcapServicesFilePath));
+            }
+            catch( final IOException e ) {
+                logger.error("Failed to read VCAP_SERVICES from file: {}", vcapServicesFilePath, e);
+                return Collections.emptyList();
+            }
```

Comment:
One could argue that this should fail, rather then returning an empty list. Because it should be reasonably unlikely that  `VCAP_SERVICES_FILE_PATH` is defined outside a scenario where the file is expected to be present.

But this library is rather lenient and also allows some IOExceptions in the other file-system-based formats, so I think for consistency it's better to apply the current solution of logging the error and return empty list here.

Reply:
I also was thinking about that but wasn't sure. I'm OK with both options but as you said for consistency we can leave it as it is at the moment.